src/keymap.c

```diff
@@ -470,6 +470,33 @@ xkb_keymap_key_for_each(struct xkb_keymap *keymap, xkb_keymap_key_iter_t iter,
         iter(keymap, key->keycode, data);
 }
 
+XKB_EXPORT xkb_keycode_t
+xkb_keymap_keyname_get_keycode(struct xkb_keymap *keymap, const char *keyname)
+{
+    struct xkb_key *key;
+    xkb_atom_t name;
+
+    name = xkb_atom_lookup(keymap->ctx, keyname);
+    if (!name)
+        return XKB_KEYCODE_INVALID;
+
+    xkb_keys_foreach(key, keymap) {
+        if (key->name == name)
+            return key->keycode;
+    }
+
+    name = XkbResolveKeyAlias(keymap, name);
```

Comment:
Would be better to canonicalize the name first, then do just one loop. An alias cannot conflict with a canonical name, so the result is the same.